glui.cpp

```diff
@@ -292,7 +356,7 @@ void glui_keyboard_func(unsigned char key, int x, int y)
       glutSetWindow( GLUI_Master.active_control_glui->get_glut_window_id() );
 
       GLUI_Master.active_control_glui->keyboard(key,x,y);
-	  finish_drawing();
+	  _sync_drawing_by_calling_glFinish();
```

Comment:
I don't know what this is about, but it creates the appears of the idle-procedure is doing the drawing. Technically I think drawing could be done inside the idle, I don't know, but it's not, so I tried to annotate it a little bit..

Note that these lines are in the "keyboard_func" here but they are also in the idle and elsewhere.

I assume this has to do with OpenGL contexts and layering, but I really have no idea. I assume at one point this solved an overdraw problem; but I would not be surprised--going by the rest of the code base--if it had never served a purpose. 

I don't believe that `glFinish` can change what appears in the frame buffer. I could be wrong, but it seems like it would only be useful for scheduling.